### PR TITLE
KTOR-4518 Add Logging for ContentNegotiation plugin

### PR DESCRIPTION
**Pull request by e5l**
Fix [KTOR-4518](https://youtrack.jetbrains.com/issue/KTOR-4518/Add-Debug-Logging-to-ContentNegotiation)

### CODE REVIEW

File: ktor-server/ktor-server-plugins/ktor-server-content-negotiation/jvmAndNix/src/io/ktor/server/plugins/contentnegotiation/RequestConverter.kt [hunk 1]
```diff
@@ -18,7 +17,12 @@ internal fun PluginBuilder<ContentNegotiationConfig>.convertRequestBody() {
         val registrations = pluginConfig.registrations
         val requestedType = call.receiveType
 
-        if (requestedType.type in pluginConfig.ignoredTypes) return@onCallReceive
+        if (requestedType.type in pluginConfig.ignoredTypes) {
+            LOGGER.trace(
+                "[ContentNegotiation] Skipping for request type ${requestedType.type} because the type is ignored."
```
Comment (whyoleg) [inline ktor-server/ktor-server-plugins/ktor-server-content-negotiation/jvmAndNix/src/io/ktor/server/plugins/contentnegotiation/RequestConverter.kt:22]: what is the reason to have `[ContentNegotiation]` in every statement, while we have logger name ended with `ContentNegotiation`?

Comment (e5l) [reply ktor-server/ktor-server-plugins/ktor-server-content-negotiation/jvmAndNix/src/io/ktor/server/plugins/contentnegotiation/RequestConverter.kt]: Fixed

**Review by rsinukov — CHANGES_REQUESTED** [review]
Please check CI

File: ktor-server/ktor-server-plugins/ktor-server-content-negotiation/jvmAndNix/src/io/ktor/server/plugins/contentnegotiation/RequestConverter.kt [hunk 2]
```diff
@@ -5,20 +5,27 @@
...
 import io.ktor.server.plugins.*
 import io.ktor.server.request.*
-import io.ktor.util.*
+import io.ktor.util.logging.*
+import io.ktor.util.reflect.*
 import io.ktor.utils.io.*
 import io.ktor.utils.io.charsets.*
 
 internal fun PluginBuilder<ContentNegotiationConfig>.convertRequestBody() {
+    val log = application.log
```
Comment (rsinukov) [inline ktor-server/ktor-server-plugins/ktor-server-content-negotiation/jvmAndNix/src/io/ktor/server/plugins/contentnegotiation/RequestConverter.kt:18]: why use application log and not create new instance as in other plugins?

**Review by rsinukov — APPROVED** [review]

File: ktor-server/ktor-server-plugins/ktor-server-content-negotiation/jvmAndNix/src/io/ktor/server/plugins/contentnegotiation/RequestConverter.kt (hunk 2 above)
Comment (e5l) [reply ktor-server/ktor-server-plugins/ktor-server-content-negotiation/jvmAndNix/src/io/ktor/server/plugins/contentnegotiation/RequestConverter.kt]: fixed

File: ktor-server/ktor-server-plugins/ktor-server-content-negotiation/jvmAndNix/src/io/ktor/server/plugins/contentnegotiation/RequestConverter.kt [hunk 3]
```diff
@@ -29,20 +34,45 @@ internal fun PluginBuilder<ContentNegotiationConfig>.convertRequestBody() {
...
-                    convertException.cause
-                )
+
+            val charset = call.request.contentCharset() ?: Charsets.UTF_8
+            for (registration in registrations) {
+                return@transformBody convertBody(body, charset, registration, requestedType, requestContentType)
+                    ?: continue
             }
+
+            LOGGER.debug("[ContentNegotiation] No suitable content converter found for request type ${requestedType.type}")
```
Comment (rsinukov) [inline ktor-server/ktor-server-plugins/ktor-server-content-negotiation/jvmAndNix/src/io/ktor/server/plugins/contentnegotiation/RequestConverter.kt:44]: `[ContentNegotiation]` is redundant

**Review by rsinukov — APPROVED** [review]

**Pull request merged**